sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/ApplyCharTypePadding.scala

```diff
@@ -39,14 +36,6 @@ import org.apache.spark.unsafe.types.UTF8String
  */
 object ApplyCharTypePadding extends Rule[LogicalPlan] {
 
-  object AttrOrOuterRef {
-    def unapply(e: Expression): Option[Attribute] = e match {
-      case a: Attribute => Some(a)
-      case OuterReference(a: Attribute) => Some(a)
-      case _ => None
-    }
-  }
-
   override def apply(plan: LogicalPlan): LogicalPlan = {
```

Comment:
Didn't refactor `apply` method since it needs to import `LogicalRelation` from `core` package. This will probably be handled like an extension in the single-pass framework